src/transformers/training_args.py

```diff
@@ -1718,7 +1718,7 @@ def _setup_devices(self) -> "torch.device":
                 )
             AcceleratorState._reset_state(reset_partial_state=True)
         self.distributed_state = None
-        if self.use_cpu:
+        if self.use_cpu or os.environ.get("ACCELERATE_USE_CPU", False):
```

Comment:
I was thinking that the training args would need to set `ACCELERATE_USE_CPU` to true for use_cpu/no cuda. Is that not the case?

Reply:
Nope it is not, as we pass cpu=True to the PartialState (as seen in the line below this)

Reply:
Thanks for explaining. I looked at the code and see now how accelerate has the shared state and global dictionary that makes this work.